### DISCUSSION TITLE
Authorization Survey

### DISCUSSION THREAD

**Discussion by alechenninger** [keycloak/keycloak#21497] [Feedback]
Hey folks, as you might've seen on the [Keycloak blog](https://www.keycloak.org/2023/07/authorization-survey), we're looking for feedback around authorization services. Reposting here for visibility:

Services dedicated to authorization are evolving rapidly. There is a steady establishment of policy languages, purpose-built for authorization, as well as a growing number of implementations of Google’s "Zanzibar" whitepaper, their [global, consistent authorization system](https://research.google/pubs/pub48190/).

To better understand how we should evolve Keycloak Authorization Services in this context, we’d appreciate the opportunity to learn more about the Keycloak communities' authorization use cases and experience, regardless of whether you’ve used Keycloak Authorization Services before or if you use a different service for access management.

Please consider filling out [this brief, anonymous survey](https://forms.gle/MkaSXQ9NuaR24qZt9) to help shape Keycloak’s future authorization experience.

**Comment by amitngm** [comment]
I have one use case on authorization. There are 2 groups. Users of both group have an aceess of role incident-user. I want to design a policy for an incident resource. User of each group can view and update a ticket for their groups only. How to achieve it

**Comment by rob-valor** [comment]
@alechenninger we are currently moving towards Keycloak because of the good documentation and extendability of the authorization services. I also filled in the survey.

Is there already more info about the future of the authorization services? It would be nice if Keycloak could support OPA as its PDP and be the PAP for OPA. So integrating OPA policies into the Keycloak UI (and gitops wise) and have the Keycloak client (PEP) call OPA server but keep the json config of it. Having a PEP driven by json config was one of the reasons we chose to us Keycloak.

**Comment by rntjr** [comment]
What will be the future of the Authorization Service, both in UI and form?
